### PR TITLE
Configure command

### PR DESCRIPTION
**Pull request by coolguru**
- support for configure
To verify
- run ‘make build’
- run ‘cloudcore-cli configure’

### CODE REVIEW

**Comment by coolguru** [comment]
TODO
- Need test PLA-1523
- Run lint and vendor

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 OSES       := linux darwin freebsd windows
 ARCHES     := 386 amd64
-EXECUTABLE ?= $(shell basename `pwd`)
+EXECUTABLE ?= cloudcoreo-cli
```
Comment (pauldallen) [inline Makefile]: lets make this "coreo"

File: cmd/configure.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var cmdConfigure = &cobra.Command{
+	Use:   CMD_CONFIG_USE,
```
Comment (pauldallen) [inline cmd/configure.go]: spacing issues

File: cmd/const.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const (
+	ACCESS_KEY = "API_KEY"
+	SECRET_KEY = "SECRET_KEY"
+	TEAM_ID = "TEAM_ID"
+	DEFAULT_FOLDER = "/.cloudcoreo"
+	DEFAULT_FILE = "profiles.yaml"
+	NONE = "None"
+
+	// root command
+	CMD_COREO_USE = "cloudcoreo"
```
Comment (pauldallen) [inline cmd/const.go]: just "coreo"

File: cmd/root.go [hunk 4]
```diff
@@ -20,20 +20,20 @@ import (
...
 var RootCmd = &cobra.Command{
-	Use:   "cloudcoreo",
-	Short: "A brief description of your application",
-	Long: `A longer description that spans multiple lines and likely contains
-examples and usage of using your application. For example:
-
-Cobra is a CLI library for Go that empowers applications.
-This application is a tool to generate the needed files
-to quickly create a Cobra application.`,
+	Use:   CMD_COREO_USE,
```
Comment (pauldallen) [inline cmd/root.go]: spacing

File: cmd/root.go [hunk 5]
```diff
@@ -55,24 +55,75 @@ func init() {
...
+	viper.AddConfigPath("$HOME/.cloudcoreo")       // adding home directory as first search path
+	viper.AutomaticEnv()               // read in environment variables that match
+
+
 	if cfgFile != "" { // enable ability to specify config file via flag
 		viper.SetConfigFile(cfgFile)
+	} else {
+		path := absPathify("$HOME")
+
+
```
Comment (pauldallen) [inline cmd/root.go]: spaces

File: cmd/const.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package cmd
+
+
+const (
+	ACCESS_KEY = "API_KEY"
+	SECRET_KEY = "SECRET_KEY"
+	TEAM_ID = "TEAM_ID"
+	DEFAULT_FOLDER = "/.cloudcoreo"
```
Comment (pauldallen) [inline cmd/const.go]: is this relative to "$HOME" ???

File: Makefile (hunk 1 above)
Comment (coolguru) [inline Makefile]: resolved

File: cmd/configure.go (hunk 2 above)
Comment (coolguru) [inline cmd/configure.go]: resovled

File: cmd/const.go (hunk 3 above)
Comment (coolguru) [inline cmd/const.go]: resolved

File: cmd/root.go (hunk 4 above)
Comment (coolguru) [inline cmd/root.go]: resolved

**Pull request merged**